x/concentrated-liquidity/msg_server_test.go

```diff
@@ -85,6 +85,73 @@ func (suite *KeeperTestSuite) TestCreateConcentratedPool_Events() {
 	}
 }
 
+// TestAddToPosition_Events tests that events are correctly emitted
+// when calling AddToPosition.
+func (suite *KeeperTestSuite) TestAddToPosition_Events() {
+	testcases := map[string]struct {
+		shouldSetupUnownedPosition   bool
+		expectedAddedToPositionEvent int
+		expectedMessageEvents        int
+		expectedError                error
+	}{
+		"happy path": {
+			shouldSetupUnownedPosition:   true,
+			expectedAddedToPositionEvent: 1,
+			expectedMessageEvents:        4,
+		},
+		"error: last position in pool": {
+			shouldSetupUnownedPosition:   false,
+			expectedAddedToPositionEvent: 0,
+			expectedError:                types.AddToLastPositionInPoolError{PoolId: 1, PositionId: 1},
+		},
+	}
+
+	for name, tc := range testcases {
+		suite.Run(name, func() {
+			suite.SetupTest()
+
+			msgServer := cl.NewMsgServerImpl(suite.App.ConcentratedLiquidityKeeper)
+
+			// Create a cl pool with a default position
+			pool := suite.PrepareConcentratedPool()
+
+			// Position from current account.
+			posId := suite.SetupDefaultPositionAcc(pool.GetId(), suite.TestAccs[0])
+
+			if tc.shouldSetupUnownedPosition {
+				// Position from another account.
+				suite.SetupDefaultPositionAcc(pool.GetId(), suite.TestAccs[1])
+			}
```

Comment:
While the "last position in pool" case uses this, I think you should just remove it as a test field (since its being used here differently than what it's original intent was)

Reply:
Good point – I changed it the test field to `lastPositionInPool` since I think we should still have a msg-level error case